### ISSUE TITLE
Automatic retopology node[request]

### ISSUE THREAD

**Issue by strike-digital** [alicevision/meshroom#774]
It would be good to have a node for automatic retopology as at the moment if you want good topology on a model you have to export it to an external program and then either bake the texture or import it back into meshroom for texturing.

I would suggest adding a node similar to the MeshDecimate node but that instead generates quad geometry on the mesh, much like the quadriflow remesher for blender or like that generated in the programme instant meshes: [https://github.com/wjakob/instant-meshes](url)
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by natowi** [comment]
Take a look at https://github.com/natowi/meshroom_external_plugins, I started a draft for instant Meshes.

**Comment by strike-digital** [comment]
Wow, that's great! But I've read the installation guide here:[https://github.com/natowi/meshroom_external_plugins/blob/master/README.md](url)
and I'm afraid it is quite hard to follow for me and I have got stuck on step 4.
I have pasted the aliceVision and qtPlugins folders into (what I think) is the correct folder but I cannot find a file named meshroom.bat in the folder or even on my whole machine. Its probably that I have just misunderstood something but thanks for your help anyway!

**Comment by natowi** [comment]
https://github.com/natowi/meshroom_external_plugins/blob/master/Meshroom.bat

**Comment by strike-digital** [comment]
Ok, thanks, that was stupid of me. I have copied the plugin files into the meshroom\nodes\aliceVision folder. On the last step, I am a little lost. I have run meshroom.bat and nothing appears to happen on the surface, so I not sure what it means by "new software". Am I missing something obvious again? Thanks.

**Comment by natowi** [comment]
In this case Instant Meshes

**Comment by strike-digital** [comment]
ah, great, and how do I run this version of meshroom then? Sorry for all the stupid questions.

**Comment by natowi** [comment]
Rename the exe to alicevision_InstantMeshes and run the bat file. Note: at the moment only one parameter (smooth) is supported.

**Comment by strike-digital** [comment]
I must have done something wrong, as all this seems to do is open the command processor for a second and then stop

**Comment by natowi** [comment]
I will update the code and pre-compile the node so it can be copied in the official binary folder.

**Comment by strike-digital** [comment]
Thank you very much. Sorry to drive you to this though.

**Comment by natowi** [comment]
Ok this can be used with the precompiled Meshroom version: Place InstantMeshes.exe in Meshroom-2019.2.0\aliceVision\bin and rename it to alicevision_InstantMeshes. 
Place the precompiled InstantMeshes.pyc in this folder: Meshroom\lib\meshroom\nodes\aliceVision

![quadrJPG](https://user-images.githubusercontent.com/8138537/73496461-f0219e80-43b8-11ea-97eb-dc528bd52fd6.JPG)

Implemented: smooth, crease, rosy, posy

**Comment by strike-digital** [comment]
That's great,Thanks!

**Comment by natowi** [comment]
[djoerg ](https://github.com/natowi/meshroom_external_plugins/issues/2#issuecomment-585736934) improved the plugin to work with the Texturing node. Set Smoothing iterations to 0 when using quad mesh.

**Comment by strike-digital** [comment]
Woah, that's really cool, thanks!

**Comment by TigerVersusT** [comment]
oh, my teacher gives me the same task!

**Comment by natowi** [comment]
@TigerVersusT you are welcome to contribute and improve the plugin

**Comment by fabiencastan** [comment]
@natowi would be great to make a PR to intergrate it when it's ready.

**Comment by TigerVersusT** [comment]
> 
> 
> @TigerVersusT you are welcome to contribute and improve the plugin

Hi,natowi! I'd like to do this, but I got some problem. Before seeing this issue, I have the following design: 1、wrap instant-meshes as a python module( been tested )
             2、import the module in app.py and export to Qt using code "self.engine.rootContext().setContextProperty("_retopology", retopo) "
             3、add a button "retopology" after  "stop" in main.qml and set "OnClicked" as call to _retopology.someFunction()
after clicking the "retopology" button, it seems that the module is not called at all ( and then I tried to change the call to member function of reconstruction added by myself which is used to open and write to a file , still nothing happened). Could you please give me some suggestion about my design, or, if it possible about how to add a new node( sorry, I'm not fully understand of the code )

**Comment by TigerVersusT** [comment]
> 
> 
> @TigerVersusT you are welcome to contribute and improve the plugin

oh, sorry for my last stupid question, I have found out the solution, by adding a thread and using @slot decorator

**Comment by natowi** [comment]
@TigerVersusT you can take a look at the latest version of the plugin https://github.com/natowi/meshroom_external_plugins/issues/2#issuecomment-595740223 and read the discussion there for details. The remaining problems are finding a way to repair self-intersections and nonmanifold faces.

**Comment by TigerVersusT** [comment]
> 
> 
> @TigerVersusT you can take a look at the latest version of the plugin [natowi/meshroom_external_plugins#2 (comment)](https://github.com/natowi/meshroom_external_plugins/issues/2#issuecomment-595740223) and read the discussion there for details. The remaining problems are finding a way to repair self-intersections and nonmanifold faces.

I've read it, you are so capable! I'll learn and see whether I can help. And thank you for all your reply.
